javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -46,8 +49,22 @@ object Math {
 
   @inline def exp(a: scala.Double): scala.Double = js.Math.exp(a)
   @inline def log(a: scala.Double): scala.Double = js.Math.log(a)
-  @inline def log10(a: scala.Double): scala.Double = log(a) / 2.302585092994046
-  @inline def log1p(a: scala.Double): scala.Double = log(a + 1)
+
+  @inline def log10(a: scala.Double): scala.Double = {
+    if (assumingES6 || !js.isUndefined(g.Math.log10))
+      js.Math.log10(a)
+    else
+      log(a) / 2.302585092994046
+  }
+
+  @inline def log1p(a: scala.Double): scala.Double = {
+    if (assumingES6 || !js.isUndefined(g.Math.log1p))
+      js.Math.log1p(a)
+    else a match {
+      case -0.0 => -0.0
```

Comment:
Pushed a fix as proposed above. Also fixed some other assertEquals on zero following the Boy Scout Rule.